tests/test/tuples.rs

```diff
@@ -34,3 +34,51 @@ fn tuple_trait_impl() {
         }
     }
 }
+
+#[test]
+fn tuples_are_sized() {
+    test! {
+        program {
+            #[lang(sized)]
+            trait Sized { }
+
+            trait Copy {}
+        }
+
+        goal {
+            (dyn Copy,): Sized
```

Comment:
This should soon not pass WF checks

Reply:
Yes, I added them so that they are actually "fixed" when tuples well-formed-ness is properly handled. This way won't forget to add such tests.

Reply:
So, we should probably rename `Copy` in these tests to something like `Foo`. Just so we don't get confused, because this is *not* the `Copy` trait.

Also, let's remove any instances where we're using `dyn Foo` where it's invalid. (i.e. not the last field of a tuple or struct). Really, we don't/aren't going to (at least not soon) check that the *usages* of types is WF, only definitions.